src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(value) {
+    if (!value.hidden && value.hasOwnProperty('hidden')) {
+      value.health -= 50;
+    }
+
+    if (value.health <= 0) {
+      Animal.alive = Animal.alive.filter((item) => item.health > 0);
```

Comment:
The filtering logic `Animal.alive.filter((item) => item.health > 0)` removes all animals with health less than or equal to zero. This logic should be executed only once when an animal's health drops to zero or below, not every time a `bite` occurs. Consider moving this logic to a separate method or adjusting the condition to ensure it only runs when necessary.